### ISSUE TITLE
How to get classes from index.ts

### ISSUE THREAD

**Issue by yantrab** [dsherret/ts-morph#558]
I wont to get all classes from one index.ts file, that look like this : 

`export * from './some-path';`

there is a way to do that?

thanks.

**Comment by dsherret** [comment]
@yantrab Try this:

```ts
import { TypeGuards } from "ts-morph";

// ...etc...

const exportedClasses = indexSourceFile.getExportedDeclarations().filter(TypeGuards.isClassDeclaration);
```

**Comment by yantrab** [comment]
thanks.
But how can i add new import for each class ?

**Comment by dsherret** [comment]
Sorry, I think I'm not understanding your question. Could you rephrase it and ask it again? Right now the original question's code has a syntax error in it. Was that meant to be `export` instead of `import`?

**Comment by yantrab** [comment]
You right, i did mistake with import instead of export.
what i wont to achieve, is to get one files, with new import from some module, and all classes from index.ts.

Thanks for you help.

**Comment by dsherret** [comment]
Could you expand on the question a bit? Maybe show a before and after of the desired code?

**Comment by yantrab** [comment]
I wont to add decorators to all classes:

```
        const project = new Project();
        const files = project.addExistingSourceFiles(modelPaths + '/**/*.ts');
        files.forEach(file =>
            file.addImportDeclaration({
                namedImports: ['IsArray', 'IsBoolean', 'IsDate', 'IsString', 'IsNumber', 'IsEnum'],
                moduleSpecifier: 'class-validator',
            }));
        const classes: ClassDeclaration[] = files
            .reduce((result, file) => result.concat(file.getClasses()), []);

        classes.forEach(c => {
            c.getProperties().forEach(prop => {
                const type = prop.getType().getText();
                const decorator = typeDecoratorMap[type];
                const isOptional = prop.hasQuestionToken();
                if (decorator) {
                    prop.addDecorator({ name: typeDecoratorMap[type], arguments: [] });
                }
            });
        });
        this.metaTypes = project.emitToMemory();
```

1. how can i using index.ts instead  `project.addExistingSourceFiles(modelPaths + '/**/*.ts')`
2. new question : how can i using model from memory?

**Comment by dsherret** [comment]
Oh ok, I think you mean something along the lines of this?

```ts
const decoratorSourceFile = ...;
const exportedClasses = indexSourceFile.getExportedDeclarations().filter(TypeGuards.isClassDeclaration);

for (const classDec of exportedClass) {
    const classSourceFile = classDec.getSourceFile();
    classSourceFile.addImportDeclaration({
        namedImports: ["YourDecoratorName"],
        moduleSpecifier: classSourceFile.getRelativePathAsModuleSpecifierTo(decoratorSourceFile)
    });
    classDec.addDecorator({ name: "YourDecoratorName" });
}
```

**Comment by dsherret** [comment]
Closing this question. Please reopen if the above does not answer it.

**Comment by adi-works** [comment]
What's the updated answer to this?

```ts
const exportedClasses = indexSourceFile.getExportedDeclarations().filter(TypeGuards.isClassDeclaration);
```

This doesn't work. `TypeGuards` is no longer exported and `getExportedDeclarations()` returns a `Map`.

Figured it out, in case anyone needs it:

```ts
function getClassesFromExports(sourceFile: SourceFile) {
  const declarations = Array.from(
    sourceFile.getExportedDeclarations().values()
  ).flat();
  const classes = declarations.flatMap((declaration) =>
    declaration
      .getSourceFile()
      .getClasses();
  );
  return classes;
}
```
